### ISSUE TITLE
[telegraf-operator] Default install does not work

### ISSUE THREAD

**Issue by nsteinmetz** [influxdata/helm-charts#86]
Hi,

Trying to discover telegraf-operator, default chart does not work:

```console
kubectl create ns telegraf-operator
helm upgrade -i telegraf-operator -n telegraf-operator influxdata/telegraf-operator
```

It leads to:

```console
NAME                                     READY   STATUS              RESTARTS   AGE
pod/telegraf-operator-55c9788d8b-lpltn   0/1     ContainerCreating   0          6m35s
pod/telegraf-operator-55c9788d8b-ppw5v   0/1     ContainerCreating   0          6m35s
pod/telegraf-operator-55c9788d8b-wq8zp   0/1     ContainerCreating   0          6m35s

NAME                        TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)   AGE
service/telegraf-operator   ClusterIP   10.3.206.109   <none>        443/TCP   6m35s

NAME                                READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/telegraf-operator   0/3     3            0           6m36s

NAME                                           DESIRED   CURRENT   READY   AGE
replicaset.apps/telegraf-operator-55c9788d8b   3         3         0       6m36s
```

Going further:

```console
kubectl describe pod/telegraf-operator-55c9788d8b-lpltn                                                                                                    
Name:           telegraf-operator-55c9788d8b-lpltn
Namespace:      telegraf-operator
Priority:       0
Node:           node-43ff5d12-3a64-432d-b16d-7dd781ad101e/51.75.199.67
Start Time:     Thu, 14 May 2020 11:14:44 +0200
Labels:         app.kubernetes.io/instance=telegraf-operator
                app.kubernetes.io/name=telegraf-operator
                pod-template-hash=55c9788d8b
Annotations:    <none>
Status:         Pending
IP:             
IPs:            <none>
Controlled By:  ReplicaSet/telegraf-operator-55c9788d8b
Containers:
  telegraf-operator:
    Container ID:  
    Image:         quay.io/influxdb/telegraf-operator:v1.0.6
    Image ID:      
    Port:          9443/TCP
    Host Port:     0/TCP
    Args:
      --telegraf-default-class=infra
      --telegraf-classes-directory=/etc/telegraf-operator
      --enable-default-internal-plugin
      --telegraf-image=docker.io/library/telegraf:1.14.1
    State:          Waiting
      Reason:       ContainerCreating
    Ready:          False
    Restart Count:  0
    Limits:
      cpu:     100m
      memory:  128Mi
    Requests:
      cpu:     50m
      memory:  64Mi
    Environment:
      POD_NAMESPACE:  telegraf-operator (v1:metadata.namespace)
    Mounts:
      /etc/certs from certs (ro)
      /etc/telegraf-operator from classes (ro)
      /var/run/secrets/kubernetes.io/serviceaccount from telegraf-operator-token-ccw57 (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  certs:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  telegraf-operator-tls
    Optional:    false
  classes:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  telegraf-operator-classes
    Optional:    false
  telegraf-operator-token-ccw57:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  telegraf-operator-token-ccw57
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason       Age                 From                                                Message
  ----     ------       ----                ----                                                -------
  Normal   Scheduled    <unknown>           default-scheduler                                   Successfully assigned telegraf-operator/telegraf-operator-55c9788d8b-lpltn to node-43ff5d12-3a64-432d-b16d-7dd781ad101e
  Warning  FailedMount  13s                 kubelet, node-43ff5d12-3a64-432d-b16d-7dd781ad101e  Unable to attach or mount volumes: unmounted volumes=[classes], unattached volumes=[certs classes telegraf-operator-token-ccw57]: timed out waiting for the condition
  Warning  FailedMount  8s (x9 over 2m16s)  kubelet, node-43ff5d12-3a64-432d-b16d-7dd781ad101e  MountVolume.SetUp failed for volume "classes" : secret "telegraf-operator-classes" not found
```

So the classe `telegraf-operator-classes`  which seems to be a secret (why not a configmap by the way ?) is not present.

It's because by default `data` is empty in `values.yml` :

```yaml
classes:
  secretName: "telegraf-operator-classes"
  default: "infra"
  data: {}
```
So it should be mentionned in the `README` that we have to provide values for data to make it work as expected.

Also on the `README`, why is it stated that `cert-manager` is required if value is false by default ? Integration with cert-manager should be better explained.

**Comment by nsteinmetz** [comment]
Seems there is an issue on accessing data from the classes file:

```
2020-05-14T13:01:38.674Z        INFO    controller-runtime.metrics      metrics server is starting to listen    {"addr": ":8080"}
2020-05-14T13:01:38.674Z        INFO    setup.entrypoint        setting up webhook server
2020-05-14T13:01:38.674Z        INFO    setup.entrypoint        registering webhooks to the webhook server
2020-05-14T13:01:38.674Z        INFO    setup.podInjector       validating class data from directory /etc/telegraf-operator
2020-05-14T13:01:38.674Z        INFO    setup.podInjector       unable to retrieve class data from file ..data: read /etc/telegraf-operator/..data: is a directory
```

**Comment by nsteinmetz** [comment]
Fixed with #92
